### PR TITLE
Add note about environment variable in parameter file.html

### PR DESCRIPTION
**Pull request by xndcn**
The default shell /bin/sh may not recognize the environment variable like ${abc.zip}
So I think we should add some notes here.

### CODE REVIEW

File: war/src/main/webapp/help/parameter/file.html
Comment (daniel-beck) [inline war/src/main/webapp/help/parameter/file.html:26]: Does this apply to all possible `/bin/sh`? Isn't the underlying implementation OS/distro-dependent?

File: war/src/main/webapp/help/parameter/file.html
Comment (daniel-beck) [inline war/src/main/webapp/help/parameter/file.html:29]: `bash` seems to be one of the shells that doesn't support it. What does doing this accomplish?

File: war/src/main/webapp/help/parameter/file.html
Comment (xndcn) [reply war/src/main/webapp/help/parameter/file.html:26]: Yes, it depends on OS/distro. So it is just a note if some one also get confused by the unrecognized environment variable `abc.zip`

On Mac OSX, /bin/sh seems link to bash, so `printenv abc.zip` returns the correct result.
But on Ubuntu 22.04, /bin/sh is a symbolic link to dash by default, which do not recognize the "abc.zip" variable.

File: war/src/main/webapp/help/parameter/file.html
Comment (xndcn) [reply war/src/main/webapp/help/parameter/file.html:29]: I confirmed `bash` can support it on my Mac OSX and Ubuntu 22.04, and the version is "3.2.57"/"5.0.17"

I use the following command to test it:
```
$ env "abc.zip=foobar" /bin/bash -c "printenv abc.zip"
foobar

$ env "abc.zip=foobar" /bin/sh -c "printenv abc.zip"
$ env "abczip=foobar" /bin/sh -c "printenv abczip"
foobar
```

Comment (daniel-beck) [reply war/src/main/webapp/help/parameter/file.html:29]: Interesting, I did not expect that to be the result. Thanks for those examples.

Comment (daniel-beck) [reply war/src/main/webapp/help/parameter/file.html:29]: I assume `dash` behaves the same after a quick look at its sources, so this could be rephrased as "Some shells, such as `dash`, or `/bin/sh` provided by `dash`, do not make such environment variables available to child processes" or similar.

(Also suggesting to replace the "recognize", while that seems to be what's happening in `dash` per `is_in_name`, it reads a bit weird from an end user POV. Thoughts?)

Comment (xndcn) [reply war/src/main/webapp/help/parameter/file.html:29]: Thank you for making it clear!

**Review by daniel-beck — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
